internal/attestation/azure/snp/validator_test.go

```diff
@@ -333,59 +370,265 @@ func TestValidateAk(t *testing.T) {
 	}
 }
 
-func int32ToByte(val uint32) []byte {
-	r := make([]byte, 4)
-	binary.PutUvarint(r, uint64(val))
-	return r
-}
+// TestGetTrustedKey tests the verification and validation of attestation report.
+func TestTrustedKeyFromSNP(t *testing.T) {
```

Comment:
It might be worth it to add a custom assert function to each testcase that checks the returned error. This will make the test a bit more brittle because we will have to do string-based testing. But it will prevent scenarios where tests that should fail, fail for the wrong reasons. E.g. because the report was malformed, but we wanted to test the signature verification.

Reply:
Added some basic assertions. For all values that alter the report and thus also result in an invalid signature, which I didn't notice before. For this, the current (placeholder) assertion is obviously not good. I will probably need to include a valid VCEK / report pair in the testdata package for each test case I guess.